datetime/datetime_test.go

```diff
@@ -73,6 +75,128 @@ func assertDatetimeIsEqual(t *testing.T, tuples []interface{}, tm time.Time) {
+		if _, ok := TimezoneToIndex[zone]; !ok {
+			t.Errorf("zone %s not found", zone)
+		}
+	}
+}
+
+func TestInvalidTimezone(t *testing.T) {
+	invalidLoc := time.FixedZone("AnyInvalid", 0)
+	tm, err := time.Parse(time.RFC3339, "2010-08-12T11:39:14Z")
+	if err != nil {
+		t.Fatalf("Time parse failed: %s", err)
+	}
+	tm = tm.In(invalidLoc)
+	dt, err := NewDatetime(tm)
+	if err == nil {
+		t.Fatalf("Unexpected success: %v", dt)
+	}
+	if err.Error() != "Unknown timezone AnyInvalid with offset 0" {
+		t.Fatalf("Unexpected error: %s", err.Error())
+	}
+}
+
+func TestInvalidOffset(t *testing.T) {
+	tests := []struct {
+		ok     bool
+		offset int
+	}{
+		{ok: true, offset: -12 * 60 * 60},
+		{ok: true, offset: -12*60*60 + 1},
+		{ok: true, offset: 14*60*60 - 1},
+		{ok: true, offset: 14 * 60 * 60},
+		{ok: false, offset: -12*60*60 - 1},
+		{ok: false, offset: 14*60*60 + 1},
+	}
+
+	for _, testcase := range tests {
+		name := ""
+		if testcase.ok {
+			name = fmt.Sprintf("in_boundary_%d", testcase.offset)
+		} else {
+			name = fmt.Sprintf("out_of_boundary_%d", testcase.offset)
+		}
+		t.Run(name, func(t *testing.T) {
+			loc := time.FixedZone("MSK", testcase.offset)
+			tm, err := time.Parse(time.RFC3339, "2010-08-12T11:39:14Z")
+			if err != nil {
+				t.Fatalf("Time parse failed: %s", err)
+			}
+			tm = tm.In(loc)
+			dt, err := NewDatetime(tm)
+			if testcase.ok && err != nil {
+				t.Fatalf("Unexpected error: %s", err.Error())
+			}
+			if !testcase.ok && err == nil {
+				t.Fatalf("Unexpected success: %v", dt)
+			}
+			if testcase.ok && isDatetimeSupported {
+				conn := test_helpers.ConnectWithValidation(t, server, opts)
+				defer conn.Close()
+
+				tupleInsertSelectDelete(t, conn, tm)
+			}
+		})
+	}
+}
+
+func TestCustomTimezone(t *testing.T) {
+	skipIfDatetimeUnsupported(t)
+
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	customZone := "Europe/Moscow"
+	customOffset := 180 * 60
+
+	customLoc := time.FixedZone(customZone, customOffset)
+	tm, err := time.Parse(time.RFC3339, "2010-08-12T11:44:14Z")
+	if err != nil {
+		t.Fatalf("Time parse failed: %s", err)
+	}
+	tm = tm.In(customLoc)
+	dt, err := NewDatetime(tm)
+	if err != nil {
+		t.Fatalf("Unable to create datetime: %s", err.Error())
+	}
+
+	resp, err := conn.Replace(spaceTuple1, []interface{}{dt, "payload"})
+	if err != nil {
+		t.Fatalf("Datetime replace failed %s", err.Error())
+	}
+	assertDatetimeIsEqual(t, resp.Data, tm)
+
+	tpl := resp.Data[0].([]interface{})
+	if respDt, ok := toDatetime(tpl[0]); ok {
+		zone, offset := respDt.ToTime().Zone()
+		if zone != customZone {
+			t.Fatalf("Expected zone %s instead of %s", customZone, zone)
+		}
+		if offset != customOffset {
+			t.Fatalf("Expected offset %d instead of %d", customOffset, offset)
+		}
+
+		_, err = conn.Delete(spaceTuple1, 0, []interface{}{dt})
+		if err != nil {
+			t.Fatalf("Datetime delete failed: %s", err.Error())
+		}
+	} else {
+		t.Fatalf("datetime doesn't match")
```

Comment:
Why do one the messages start with uppercase and another one with the lowercase?

Reply:
Thank you, fixed!